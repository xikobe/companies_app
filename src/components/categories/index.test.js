import React from 'react';
import { shallow } from 'enzyme';
import { useSelector } from 'react-redux';
import Categories from './index';
import CategoryItem from './category-item';

jest.mock('react-redux');

const mockCategories = ['Foo', 'Bar'];
useSelector.mockImplementation(() => mockCategories);

describe('Categories', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Categories />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CategoryItem)).toHaveLength(mockCategories.length);
    expect(wrapper.find(CategoryItem).map((node) => node.props()))
      .toEqual(mockCategories.map((cat) => ({ name: cat })));
  });
});
