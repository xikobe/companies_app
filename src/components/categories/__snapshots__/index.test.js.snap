// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<Styled(nav)>
  <h2>
    Categories
  </h2>
  <ul>
    <CategoryItem
      key="Foo"
      name="Foo"
    />
    <CategoryItem
      key="Bar"
      name="Bar"
    />
  </ul>
</Styled(nav)>
`;
